### PR TITLE
make test harness importable

### PR DESCRIPTION
**Pull request by Shell32-Natsu**
Make the e2e test harness importable so it can be reused in kpt pipeline e2e test.

@natasha41575

### CODE REVIEW

File: tests/e2etest/tests/e2e_test.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package e2etest
 import (
 	"testing"
 
-	"github.com/GoogleContainerTools/kpt-functions-catalog/tests/e2etest/internal/runner"
```
Comment (natasha41575) [inline tests/e2etest/tests/e2e_test.go:6]: is it possible to fix the typo that causes the difference between the name of the module here and the name of the directory?
https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/tests/e2e-test/go.mod

Comment (Shell32-Natsu) [reply tests/e2etest/tests/e2e_test.go:6]: The name of the directory has been updated.

Comment (natasha41575) [reply tests/e2etest/tests/e2e_test.go:6]: Ah, my mistake, I didn't see that. Thank you

**Comment by mengqiy** [comment]
I expect the end states are:
- The e2e test harness in the kpt-functions-catalog repo is merged with the test harness in the kpt repo.
- The test harness in the kpt repo is stable enough. IOW, it is not updated very often.
- kpt-functions-catalog repo depends on the test harness in the kpt repo for e2e tests.

Making it exportable can be the 1st step to transition to the end states mentioned above.

**Review by mengqiy — APPROVED** [review]

**Pull request merged**